### PR TITLE
Allows skipping specific levels of individual checks

### PR DESCRIPTION
**Pull request by benjwadams**
Allows skipping of specific priority levels from within checks by using
a modified form of `<skip_check>:<max_level>`.  Max level can be one of
"A", "M", or "L", which correspond to skipping all output (same as
default invocation with `<skip_check>`), skipping medium and low priority
items within the check, and skipping only low priority checks,
respectively.  Implements #589.

### CODE REVIEW

**Comment by Bobfrat** [comment]
@mhidas FYI

**Comment by benjwadams** [comment]
@mhidas, tried to add you as a reviewer here, but I can't.  Please give me any feedback on the implementation you have.

**Comment by benjwadams** [comment]
Is this OK?  Can we merge?

**Comment by Bobfrat** [comment]
I'm still a little confused about the use case. @mhidas does this suit your needs?

**Comment by mhidas** [comment]
Apologies for the slow response. Yes, this does exactly what I was hoping for. Thanks!

File: compliance_checker/suite.py [hunk 1]
```diff
@@ -122,16 +125,36 @@ def _get_checks(self, checkclass, skip_checks):
...
-    def _run_check(self, check_method, ds):
+        return returned_checks
+
+    def _run_check(self, check_method, ds, max_level):
+        """
+        Runs a check and appends a result to the
+        """
         val = check_method(ds)
 
         if isinstance(val, list):
```
[the remark this answers is not in the document]
Comment (benjwadams) [reply compliance_checker/suite.py]: I probably should just track down wherever this is in the code and pass single elements as a one element list.

**Comment by benjwadams** [comment]
OK, going to merge.  @mhidas, did see your comment -- I'd rather convert the single object case to a list earlier in the code to make the handling consistent.

**Pull request merged**